sympy/solvers/diophantine.py

```diff
@@ -875,9 +875,9 @@ def _diop_quadratic(var, coeff, t):
                     + (sqa*g*u**2 + D*u + sqa*F) // _c
 
                 for z0 in range(0, abs(_c)):
-                    if divisible(
-                            sqa*g*z0**2 + D*z0 + sqa*F,
-                            _c):
+                    _temp = sqa*g*z0**2 + D*z0 + sqa*F
+                    # Check if the values of y and x obtained are integers or not
+                    if divisible(_temp, _c) and divisible(_temp*e*sqc - z0*_c, sqa*_c):
```

Comment:
In line 880 of the code visible in the snippet, observe the `if` statement which is
`if divisible(_temp, _c) and divisible(_temp*e*sqc - z0*_c, sqa*_c):`
Here there are two conditions
1. y is integral iff `divisible(_temp, _c)`
2. x is integral iff `divisible(_temp*e*sqc - z0*_c, sqa*_c)`
So the condition for x and y to be integral is already present in the code